File: z_cs_py_yapilanlar/CTxtFileWriter.cs
public class CTxtFileWriter
{
    string FileName;
    FileStream fs;
    StreamWriter sw;
    bool IsOpenedFlag;

    public CTxtFileWriter()
    {
        IsOpenedFlag = false;
    }

    public bool OpenFile(string fileName, bool appendEnabled = false)
    {
        fs = new FileStream(fileName, appendEnabled ? FileMode.Append : FileMode.Create, FileAccess.Write, FileShare.ReadWrite);
        sw = new StreamWriter(fs);

        IsOpenedFlag = true;
        return IsOpenedFlag;
    }

    public bool CloseFile()
    {
        sw.Close();
        fs.Close();

        IsOpenedFlag = false;
        return IsOpenedFlag;
    }

    public bool IsOpened()
    {
        return this.IsOpenedFlag;
    }

    public void WriteLine(string text)
    {
        sw.WriteLine(text);
        sw.Flush();
    }

    public void Write(string text)
    {
        sw.Write(text);
        sw.Flush();
    }

    private void WriteData()
    {
        FileStream fs = new FileStream("c:\\test.txt", FileMode.Append, FileAccess.Write);
        StreamWriter sw = new StreamWriter(fs);
        Console.WriteLine("Enter the text which you want to write to the file");
        string str = Console.ReadLine();
        sw.WriteLine(str);
        sw.Flush();
        sw.Close();
        fs.Close();
    }
}

File: z_cs_py_yapilanlar/CTxtFileWriter.py
class CTxtFileWriter:
    def __init__(self):
        self.FileName = None
        self.fs = None
        self.sw = None
        self.IsOpenedFlag = False

    def open_file(self, file_name, append_enabled=False):
        mode = 'a' if append_enabled else 'w'
        self.fs = open(file_name, mode)
        self.IsOpenedFlag = True
        return self.IsOpenedFlag

    def close_file(self):
        if self.fs:
            self.fs.close()
        self.IsOpenedFlag = False
        return self.IsOpenedFlag

    def is_opened(self):
        return self.IsOpenedFlag

    def write_line(self, text):
        if self.fs:
            self.fs.write(text + '\n')
            self.fs.flush()

    def write(self, text):
        if self.fs:
            self.fs.write(text)
            self.fs.flush()

    def write_data(self):
        with open('c:\\test.txt', 'a') as f:
            text_to_write = "This is the text to be written to the file."
            f.write(text_to_write + '\n')
